src/libraries/System.Formats.Tar/src/System/Formats/Tar/PosixTarEntry.cs

```diff
@@ -0,0 +1,102 @@
+    /// </summary>
+    /// <remarks>Formats that implement the POSIX IEEE P1003.1 standard from 1988, support the following header fields: <c>devmajor</c>, <c>devminor</c>, <c>gname</c> and <c>uname</c>.
+    /// Even though the <see cref="TarFormat.Gnu"/> format is not POSIX compatible, it implements and supports the Unix-specific fields that were defined in that POSIX standard.</remarks>
+    public abstract partial class PosixTarEntry : TarEntry
+    {
+        // Constructor used when reading an existing archive.
+        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin)
+            : base(header, readerOfOrigin)
+        {
+        }
+
+        // Constructor called when creating a new 'TarEntry*' instance that can be passed to a TarWriter.
+        internal PosixTarEntry(TarEntryType entryType, string entryName, TarFormat format)
+            : base(entryType, entryName, format)
+        {
+        }
+
+        /// <summary>
+        /// When the current entry represents a character device or a block device, the major number identifies the driver associated with the device.
+        /// </summary>
+        /// <remarks>Character and block devices are Unix-specific entry types.</remarks>
+        /// <exception cref="InvalidOperationException">The entry does not represent a block device or a character device.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Cannot set a negative value.</exception>
+        public int DeviceMajor
+        {
+            get => _header._devMajor;
+            set
+            {
+                if (_header._typeFlag is not TarEntryType.BlockDevice and not TarEntryType.CharacterDevice)
+                {
+                    throw new InvalidOperationException(SR.TarEntryBlockOrCharacterExpected);
+                }
+
+                if (value < 0 || value > 99_999_999)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+                _header._devMajor = value;
+            }
+        }
+
+        /// <summary>
+        /// When the current entry represents a character device or a block device, the minor number is used by the driver to distinguish individual devices it controls.
+        /// </summary>
+        /// <remarks>Character and block devices are Unix-specific entry types.</remarks>
+        /// <exception cref="InvalidOperationException">The entry does not represent a block device or a character device.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Cannot set a negative value.</exception>
+        public int DeviceMinor
```

Comment:
Subsequently, would it make sense for DeviceMajor/Minor to be combined into a single property? e.g. of type `(int Major, int Minor)`, or even `Version`?

Reply:
We can discuss this in a separate issue to determine if we want to modify these public APIs into what you proposed.